# In-Class Activity #6 (July 23, 2026)

In [1]:
library(tidyverse)
library(moderndive)
library(car)

── Attaching core tidyverse packages ─────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ───────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Loading required package: carData


Attaching package: ‘car’


The following object is masked from ‘package:dplyr’:

    recode


The following object is masked from ‘package:purrr’:

    some




In [2]:
wage_data <- read.table("data/wage.txt", header=TRUE)
head(wage_data)

,education,south,sex,experience,union,wage,age,race,occupation,sector,marr
,<int>,<int>,<int>,<int>,<int>,<dbl>,<int>,<int>,<int>,<int>,<int>
1,8,0,1,21,0,5.10,35,2,6,1,1
2,9,0,1,42,0,4.95,57,3,6,1,1
3,12,0,0,1,0,6.67,19,3,6,1,0
4,12,0,0,4,0,4.00,22,3,6,0,0
5,12,0,0,17,0,7.50,35,3,6,0,1
6,13,0,0,9,1,13.07,28,3,6,0,0


### Part A

In [3]:
wage_data$wage_categorical <- as.numeric(wage_data$wage > mean(wage_data$wage))
head(wage_data)

,education,south,sex,experience,union,wage,age,race,occupation,sector,marr,wage_categorical
,<int>,<int>,<int>,<int>,<int>,<dbl>,<int>,<int>,<int>,<int>,<int>,<dbl>
1,8,0,1,21,0,5.10,35,2,6,1,1,0
2,9,0,1,42,0,4.95,57,3,6,1,1,0
3,12,0,0,1,0,6.67,19,3,6,1,0,0
4,12,0,0,4,0,4.00,22,3,6,0,0,0
5,12,0,0,17,0,7.50,35,3,6,0,1,0
6,13,0,0,9,1,13.07,28,3,6,0,0,1


In [4]:
wage_logistic_model <- glm(formula = wage_categorical ~ education, data=wage_data, family='binomial')
summary(wage_logistic_model)


Call:
glm(formula = wage_categorical ~ education, family = "binomial", 
    data = wage_data)

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  -4.4109     0.5549  -7.949 1.88e-15 ***
education     0.3017     0.0410   7.359 1.86e-13 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 716.62  on 533  degrees of freedom
Residual deviance: 652.30  on 532  degrees of freedom
AIC: 656.3

Number of Fisher Scoring iterations: 3


`Education` is associated with an individual's wage being above/below average given the very small p-value `1.86e-13`. The `0.3017` estimate indicates that each additional year of education is associated with multiplying the odds (of being above average) by `e^0.3017 = 1.35`. This means that each additional year of education is associated with a 35% increase in the odds that the wage will be above average.